src/test/scala/org/broadinstitute/dsde/firecloud/service/NihServiceSpec.scala

```diff
@@ -47,4 +52,31 @@ class NihServiceSpec extends AnyFlatSpec with Matchers {
+    val pubKey = "-----BEGIN PUBLIC KEY-----\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA41cb+haTw2qFQk7SF1ge\nFNF62BgmAvOnVW3IvK/RCjryak5hLL0mwZcekRcq91bQQdlOlqMKR4poNZdHidXE\nkCsCdWcnftbY7+U3rZ4zIYLeWl7M7tT+NvPVC+h8H3BBUl7dyy9WtAOMe3oiZF5y\nm9MuNCBaOepSJ3jPeIHI7mUpYfo+ivyS03HPmZX8E3R05PGrHPvzcDafyd139KA+\n8l4cwHV2yYOQAr3AZO6XzCIcyT3q6CtQ4EiBuBOYVK9j/7Tbgq2aLS92QY8sht8e\nDhohhu+zGv376qIVsDTLP1t+PTooUKXVltxPjvSfjQS++7C8YlDKdZQB+5MdL2iu\nXQIDAQAB\n-----END PUBLIC KEY-----"
+    val mockShibboleth = mock[ShibbolethDAO]
+    when(mockShibboleth.getPublicKey()).thenReturn(Future.successful(pubKey))
+    val nihServiceMock = new NihService(samDao, thurloeDao, googleDao, mockShibboleth)
+
+    //expires in the year 2052
+    val validStr = Jwt.encode(JwtClaim("{\"eraCommonsUsername\": \"firecloud-dev\", \"iat\": 1652937842}").expiresAt(2602002551L), privKey, JwtAlgorithm.RS256)
+    val validJwt = JWTWrapper(validStr)
+
+    //expired in June 2022
+    val expStr = Jwt.encode(JwtClaim("{\"eraCommonsUsername\": \"firecloud-dev\", \"iat\": 1655232707}").expiresAt(1655232707), privKey, JwtAlgorithm.RS256)
+    val expJwt = JWTWrapper(expStr)
+
+    val userToken: UserInfo = UserInfo("dummyToken", thurloeDao.TCGA_AND_TARGET_LINKED)
+
+    val resp1 = Await.result(nihServiceMock.updateNihLinkAndSyncSelf(userToken, validJwt), 3.seconds)
+    val resp2 = Await.result(nihServiceMock.updateNihLinkAndSyncSelf(userToken, expJwt), 3.seconds)
+
+    assert(resp1.toString.contains("200 OK"), "Unexpired token should be accepted")
+    assert(resp2.toString.contains("400 Bad Request"), "Expired token should be rejected")
+    assert(resp2.toString.contains("Failed to decode JWT"), "Expired token should fail at the decode stage")
```

Comment:
using `toString.contains` here feels somewhat inexact. You could be a bit more typesafe and precise using Scala pattern matching - which is, I admit, uglier and more complex:

```scala
    resp1 match {
      case _@ RequestComplete((StatusCodes.OK, _)) => succeed
      case x => fail(s"Unexpired token should be accepted. Response was: $x")
    }

    resp2 match {
      case _@ RequestComplete((StatusCodes.BadRequest, errorReport: ErrorReport)) =>
        errorReport.message shouldBe "Failed to decode JWT"
      case x =>
        fail(s"Expired token should fail at the decode stage. Response was: $x")
    }
```

Reply:
I messed around a lot with how to check this - I think I just don't know Scala well enough to figure out how to deal with the typing here, so thank you!